src/js2dt.js

```diff
@@ -0,0 +1,352 @@
+  }
+  try {
+    var emitter = new RAMLEmitter(loadJSONFile(fileName), typeName)
+    var ramledData = emitter.emit()
+  } catch (error) {
+    cb(error, null)
+    return
+  }
+  cb(null, yaml.safeDump(ramledData, {'noRefs': true}))
+}
+
+/**
+ * Holds functions to convert JSON schema to RAML data type.
+ *
+ * @param  {Object} data
+ * @param  {string} typeName - Name of RAML data type to hold converted data.
+ */
+function RAMLEmitter (data, typeName) {
+  this.data = data
+  this.mainTypeName = typeName
+  this.types = {}
+
+  /**
+   * Process definitions, delete them from data and add processed
+   * values to `this.types`.
+   */
+  this.processDefinitions = function () {
+    var defsData = this._processDefinitions(this.data.definitions)
+    delete this.data.definitions
+    this.types = utils.updateObjWith(this.types, defsData)
+  }
+
+  /**
+   * Process main data and save result in `this.types`.
+   */
+  this.processMainData = function () {
+    delete this.data['$schema']
+    this.types[this.mainTypeName] = this.ramlForm(this.data, [])
+  }
+
+  /**
+   * Run conversion process and emit results.
+   */
+  this.emit = function () {
+    this.processDefinitions()
+    this.processMainData()
+    return {'types': this.types}
+  }
+
+  /**
+   * Convert JSON schema to a library with RAML data type.
+   *
+   * @param  {Object} data - Data to be converted.
+   * @param  {Array} reqStack - Stack of required properties.
+   * @param  {string} [prop] - Property name nested objects of which are processed.
+   * @returns  {Object}
+   */
+  this.ramlForm = function (data, reqStack, prop) {
+    if (!(data instanceof Object)) {
+      return data
+    }
+    var isObj = data.type === 'object'
+    if (isObj) {
+      reqStack.push({
+        'reqs': data.required || [],
+        'props': Object.keys(data.properties || {})
+      })
+      delete data.required
+    }
+
+    var combsKey = getCombinationsKey(data)
+    if (combsKey && data.type) {
+      data = addCombinationsType(data, combsKey)
+    }
+
+    if (isFileType(data)) {
+      data = changeFileType(data)
+    }
+
+    var updateWith = this.processNested(data, reqStack)
+    data = utils.updateObjWith(data, updateWith)
+
+    if (isObj) {
+      reqStack.pop()
+    }
+    var lastEl = reqStack[reqStack.length - 1]
+    if (lastEl && prop) {
+      if (lastEl.props.indexOf(prop) > -1) {
+        data['required'] = lastEl.reqs.indexOf(prop) > -1
+      }
+    }
+
+    if (data['$ref']) {
+      data = replaceRef(data)
+    } else if (data.type) {
+      data = changeType(data)
+      data = changeDateType(data)
+    }
+    if (combsKey) {
+      data = this.processCombinations(data, combsKey, prop)
+    }
+    return data
+  }
+
+  /**
+   * Process definitions the same way main type is processed
+   * (using ramlForm).
+   *
+   * @param  {Object} defs - JSON 'definitions' object.
+   * @returns  {Object}
+   */
+  this._processDefinitions = function (defs) {
```

Comment:
It was originally called `processDefinitions` but then it became part of new `processDefinitions` func so I just added `_`. Renamed in https://github.com/raml-org/ramldt2jsonschema/commit/821de8951217a26a91b941b6f13e67d018c3d848